src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+import {Position} from "./position"
+import {Probability} from "./probability"
+
+export class GameState{
```

Comment:
This is more like it!

This is a great example of separating out a class to simplify the rest of your code!